ambry-network/src/main/java/com/github/ambry/network/LocalRequestResponseChannel.java

```diff
@@ -125,6 +125,8 @@ static ByteBuf byteBufFromPayload(Send payload) throws IOException {
     payload.writeTo(new ByteBufferChannel(ByteBuffer.wrap(SIZE_BYTE_ARRAY)));
     WritableByteChannel byteChannel = Channels.newChannel(new ByteBufOutputStream(buffer));
     payload.writeTo(byteChannel);
+    // Just like MockSelector, LocalRequestResponseChannel needs to release the sends' resources after completed.
+    payload.release();
```

Comment:
I don't think this is the best place to release the payload, I would recommend releasing the payload in the caller methods instead of this method.

Reply:
Thanks for the comment.

We have another PR which avoids serialization of requests when using LocalNetworkClient as below. 
https://github.com/linkedin/ambry/pull/1981 
https://github.com/linkedin/ambry/pull/1981/files#diff-f3bc94decabe1a16937bcab2ca59e84358d8e472facd2044745f84dffb46dbfe

Depending on the new code, the problem could still exist or don't. Need verify the memory leak with the new code.

So will onhold this PR.